UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  OTCTextField.swift
+//  
+//
+//  Created by Ahmed Nagy on 08/07/2022.
+//
+
+import UIKit
+
+class OTCTextField: UITextField {
+    
+    var didEnterLastDigit: ((String) -> Void)?
+    var defaultCharacter = "-"
+    private var isConfigured = false
+    private var digitsLabels = [UILabel]()
+    private lazy var tapRecognizer: UITapGestureRecognizer = {
+        let recognizer = UITapGestureRecognizer()
+        recognizer.addTarget(self, action: #selector(becomeFirstResponder))
+        return recognizer
+    }()
+    
+    func configure(with slotCount: Int = 4) {
+        guard isConfigured == false else { return }
+        isConfigured.toggle()
+        configureTextField()
+        let labelsStackView = createLabelStackView(with: slotCount)
+        addSubview(labelsStackView)
+        addGestureRecognizer(tapRecognizer)
+        NSLayoutConstraint.activate([
+            labelsStackView.topAnchor.constraint(equalTo: topAnchor),
+            labelsStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            labelsStackView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            labelsStackView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        ])
+    }
+    private func configureTextField() {
+        tintColor = .clear
+        textColor = .clear
+        keyboardType = .numberPad
+        textContentType = .oneTimeCode
+        addTarget(self, action: #selector(textDidChange), for: .editingChanged)
+        delegate = self
+    }
+    private func createLabelStackView(with count: Int) -> UIStackView {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.alignment = .fill
+        stackView.distribution = .fillEqually
+        stackView.spacing = 8
```

Comment:
Configuration which might change per class are preferably to be added in enum in the class 

For example
```swift 
extension OTCTextField { 
    
    enum Metrics { 
        static let labelsSpacing: CGFloat = 8.0
    }

    enum Style { 
        static let font: UIFont = .systemFont(ofSize: 40)
    }
}
```

Should be aded at the end of the file.